projects/step-frontend/src/lib/modules/function/components/function-package-configuration-dialog/function-package-configuration-dialog.component.ts

```diff
@@ -0,0 +1,153 @@
+          return of(false);
+        }),
+        tap(() => (this.isLoading = false))
+      )
+      .subscribe((result) => this._matDialogRef.close(result));
+  }
+
+  cancel(): void {
+    this._resourceInputBridgeService.deleteLastUploadedResource();
+    this._matDialogRef.close(false);
+  }
+
+  addRoutingCriteria(): void {
+    this.criteria.push({ key: '', value: '' });
+  }
+
+  saveRoutingCriteria(): void {
+    this.functionPackage.tokenSelectionCriteria = this.criteria
+      .filter((item) => !!item.key)
+      .reduce((res, item) => {
+        res[item.key] = item.value;
+        return res;
+      }, {} as Record<string, string>);
+  }
+
+  removeRoutingCriteria(criterion: KeyValue<string, string>): void {
+    if (this.functionPackage.tokenSelectionCriteria) {
+      delete this.functionPackage.tokenSelectionCriteria[criterion.key];
+    }
+    this.criteria = this.criteria.filter((item) => item !== criterion);
+  }
+
+  saveCustomAttributes(customAttributes?: Record<string, unknown>): void {
+    this.customAttributes = customAttributes as FunctionPackageConfigurationDialogComponent['customAttributes'];
+    this.functionPackage.packageAttributes = this.customAttributes.attributes;
+  }
+
+  handlePackageLocationChange(packageLocation?: string): void {
+    this.functionPackage.packageLocation = packageLocation;
+    this.loadPackagePreview();
+  }
+
+  handlePackageLibrariesLocationChange(packageLibrariesLocation?: string): void {
+    this.functionPackage.packageLibrariesLocation = packageLibrariesLocation;
+    this.loadPackagePreview();
+  }
+
+  private loadPackagePreview(): void {
+    of(undefined)
+      .pipe(
+        tap(() => {
+          this.isLoading = true;
+          this.isFunctionPackageReady = false;
+          this.previewError = undefined;
+          this.addedFunctions = undefined;
+        }),
+        switchMap(() =>
+          iif(
+            () => !this.functionPackage?.packageLocation,
+            of(undefined),
+            this._api.packagePreview(this.functionPackage).pipe(
+              map((response) => {
+                const addedFunctions = response.functions;
+                const previewError = response.loadingError;
+                return { addedFunctions, previewError };
+              }),
+              catchError((error) =>
+                of({
+                  previewError: error.data,
+                  addedFunctions: undefined,
+                })
+              )
+            )
+          )
+        ),
+        tap(() => (this.isLoading = false))
+      )
+      .subscribe((result) => {
+        this.addedFunctions = result?.addedFunctions;
+        this.previewError = result?.previewError;
+        if (this.previewError) {
+          return;
+        }
+        if (this.addedFunctions?.length) {
+          this.isFunctionPackageReady = true;
+        } else {
+          this.previewError = `No keyword were found!`;
```

Comment:
I see you are using ` to mark strings, I usually use '. Lets align on that in the weekly